test/e2e/apimachinery/flowcontrol.go

```diff
@@ -324,6 +345,35 @@ func createFlowSchema(f *framework.Framework, flowSchemaName string, matchingPre
 	}
 }
 
+// waitForSteadyState repeatedly polls the API server to check if the newly
+// created flow schema and priority level have been seen by the APF controller
+// by checking: (1) the dangling priority level reference condition in the flow
+// schema status, and (2) metrics. The function times out after 30 seconds.
+func waitForSteadyState(f *framework.Framework, flowSchemaName string, priorityLevelName string) {
```

Comment:
Returning an error from this function and letting callers to handle them as they want would be a bit cleaner.

Reply:
What is meant by "cleaner" here?  The suggested change would make the code more complex.

Reply:
Will make it a bit longer - I don't agree it's more complex.

I actually think that if something can return an error, I want to leave the decision to the called how they handle it, rather the hiding that behavior below.
As an example - the error returned FlowSchemas.Get() - I can easily imagine a caller may want to retry that in some cases. Another example is that I may want to completely differently handle the case of error from API call and differently a timeout from waiting for reconciliation.
Making a decision for the called how they should do that is not the best one.

Reply:
I understand that kind of thinking for non-test code.  For test code, it seems like unnecessary generality to me --- we do not have an open world of code that will be calling the test code.

I am not sure what complexity measure you are thinking of.  I think it is simpler to wrap up things as they are.  Returning an error requires duplicating the error-handling at all the call sites, and in this case they all want the same handling.  It is only test code; if some day we find the test code wants different error handling in different tests, we can refactor then.

But this is a pretty small detail; I can live with the suggested change.

Reply:
I think it's better to change, but it's not something I'm going to fight until death too :)

Reply:
Another way to do this is to define the checking function, and let the caller invoke via the `framework.ExpectNoError(wait.Poll(...))` line. Given the number and uniformity of the invocations, though, I think that's not necessary yet.

We can always change it later, let's leave it for now.